File: app/src/main/java/com/left/gank/ui/android/AndroidContract.java
package com.left.gank.ui.android;

import android.content.Context;
import android.ly.business.domain.Gank;

import androidx.annotation.NonNull;

import com.left.gank.mvp.base.ObserverPresenter;
import com.left.gank.mvp.base.SupportView;

import java.util.List;

/**
 * Create by LingYan on 2016-10-25
 */

public interface AndroidContract {
    interface View extends SupportView {
        void loadAndroidSuccess(int page, List<Gank> list);

        void loadAndroidFailure(int page, String msg);
    }

    abstract class Presenter extends ObserverPresenter<View> {

        public Presenter(@NonNull Context context, @NonNull View view) {
            super(context, view);
        }

        public abstract void loadAndroid(boolean refresh, boolean useProgress, int page);
    }
}

File: app/src/main/java/com/left/gank/ui/android/AndroidContract.kt
package com.left.gank.ui.android

import android.business.domain.Gank
import android.content.Context

import com.left.gank.mvp.base.ObserverPresenter
import com.left.gank.mvp.base.SupportView

/**
 * Create by LingYan on 2016-10-25
 */

interface AndroidContract {
    interface View : SupportView {
        fun loadAndroidSuccess(page: Int, list: List<Gank>)

        fun loadAndroidFailure(page: Int, msg: String)
    }

    abstract class Presenter(context: Context, view: View) : ObserverPresenter<View>(context, view) {

        abstract fun loadAndroid(refresh: Boolean, useProgress: Boolean, page: Int)
    }
}
